### PR TITLE
Document `CellType` enum members with figures

### LINKED ISSUE #6549: Add a figure of CellTypes to explain the shape and the order of node in `pyvista.CellType` section
### Describe what maintenance you would like added.

We a figure of CellTypes to explain the shape and the order of node in [`pyvista.CellType` section](https://docs.pyvista.org/api/utilities/_autosummary/pyvista.celltype#pyvista-celltype).

### Links to source code.

https://docs.pyvista.org/api/utilities/_autosummary/pyvista.celltype#pyvista-celltype

### Pseudocode or Screenshots

image

### PR DESCRIPTION
**Pull request by user27182**
### Overview

- [x] Allow documenting individual enum members (see https://enum-tools.readthedocs.io/en/latest/api/autoenum.html)
- [x] Add cell info
- [x] Add figures for linear cells
- [x] Format table 

Resolve #6549

The docstring text for the cells is adapted from each of the vtk docstrings for the respective cell, as well as information from the VTK book: https://book.vtk.org/en/latest/VTKBook/05Chapter5.html#

### CODE REVIEW

**Comment by user27182** [comment]
Still a few kinks to work out but here's a preview of the new docs. (Note the position of image and text will be swapped).
celltypes

**Comment by user27182** [comment]
Looks good on mobile (portrait and landscape)
image
image

**Comment by tkoyama010** [comment]
Thank you! This is exactly what I want to.

**Comment by user27182** [comment]
pre-commit.ci autofix

**Comment by user27182** [comment]
Can't decide on the styling: should the points, edges, faces have black text with white fill (bottom), or white text with black fill (top)?
muted-line

muted

**Comment by tkoyama010** [comment]
+1 black text with white fill (bottom)

File: pyvista/core/celltype.py [hunk 1]
```diff
@@ -3,18 +3,244 @@
...
+    cell_class: type[_vtk.vtkCell] | None = None
+    short_doc: str | None = None
+    long_doc: str | None = None
+    example: str | None = None
+    points_override: Literal['variable', 'n/a'] | None = None
+    edges_override: Literal['variable', 'n/a'] | None = None
+    faces_override: Literal['variable', 'n/a'] | None = None
+
+
+class _DocIntEnum(IntEnum):
```
Comment (user27182) [inline pyvista/core/celltype.py:94]: I think we can get rid of this class, it only contains `CellType` specific code and can't really be reused. Just call it `CellType`  instead

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**